lib/va_profile/models/service_history.rb

```diff
@@ -80,6 +80,30 @@ def self.build_from_academy_episode(episode)
           end_date: episode['academy_end_date']
         )
       end
+
+      def self.determine_eligibility(episodes)
+        problem_message = [
+          'We’re sorry. There’s a problem with your discharge status records. We can’t provide a Veteran status ' \
+          'card for you right now.',
+          'To fix the problem with your records, call the Defense Manpower Data Center at 800-538-9552 (TTY: 711).' \
+          ' They’re open Monday through Friday, 8:00 a.m. to 8:00 p.m. ET.'
+        ]
+        not_eligible_message = [
+          'Our records show that you’re not eligible for a Veteran status card. To get a Veteran status card, you ' \
+          'must have received an honorable discharge for at least one period of service.',
+          'If you think your discharge status is incorrect, call the Defense Manpower Data Center at 800-538-9552 ' \
+          '(TTY: 711). They’re open Monday through Friday, 8:00 a.m. to 8:00 p.m. ET.'
+        ]
+
+        return { confirmed: false, message: problem_message } if episodes.empty?
+
+        codes = episodes.map(&:character_of_discharge_code).uniq.compact
+        return { confirmed: true, message: [] } if codes.intersect?(%w[A B H J]) # Honorable discharge
+        # Not honorable discharge
+        return { confirmed: false, message: not_eligible_message } if codes.intersect?(%w[D E F K]) || codes.empty?
```

Comment:
😎

Reply:
It occurs to me that maybe you might also want to move these to constants on the service object eg `HONORABLE_DISCHARGE_CODES`, `DISHONORABLE_DISCHARGE_CODES` 

Again, not a blocker.